### PR TITLE
[quant][graphmode] Fix a corner case in handling `if` in insert_observers

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #39645 [quant][graphmode] Support quantized::batch_norm2d_relu fusion for tracing
* #39644 [quant][graphmode] Support quantization for `aten::append`
* **#39615 [quant][graphmode] Fix a corner case in handling `if` in insert_observers**

Summary:
corner case: https://github.com/pytorch/vision/blob/master/torchvision/models/densenet.py#L87

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21942110](https://our.internmc.facebook.com/intern/diff/D21942110)

### CODE REVIEW

File: test/quantization/test_quantize_script.py [hunk 1]
```diff
@@ -751,7 +751,33 @@ def forward(self, x):
...
+            def __init__(self, cond):
+                super(M, self).__init__()
+                self.conv = torch.nn.Conv2d(3, 3, 3).float()
+                self.cond = cond
+
+            def forward(self, x):
+                x = self.conv(x)
+                # x is already observed
+                if self.cond:
+                    x = torch.flatten(x)
```
Comment (raghuramank100) [inline test/quantization/test_quantize_script.py:769]: If this operation where instead 
```python
if self.cond:
  x = self.conv1(x)
return x
```
you would need to insert an observer  after conv1. Is this tested?

**Review by raghuramank100 — COMMENTED** [review]
A question on test coverage.

File: test/quantization/test_quantize_script.py (hunk 1 above)
Comment (jerryzh168) [reply test/quantization/test_quantize_script.py:769]: not yet, I can add the test

Comment (jerryzh168) [reply test/quantization/test_quantize_script.py:769]: this does not work since different branches of `if` not quantized the same way

Comment (jerryzh168) [reply test/quantization/test_quantize_script.py:769]: there is a conflict of rules at the moment if we want to support this, we can probably think about it later

Comment (raghuramank100) [reply test/quantization/test_quantize_script.py:769]: Please file an issue to track this issue down the road.

**Review by raghuramank100 — APPROVED** [review]

**Pull request closed without merging**